addons/survey/models/survey_user.py

```diff
@@ -379,17 +424,3 @@ def _check_answer_type_skipped(self):
 
             if field_name and not line[field_name]:
                 raise ValidationError(_('The answer must be in the right type'))
-
-    @api.model_create_multi
-    def create(self, vals_list):
-        for vals in vals_list:
-            suggested_answer_id = vals.get('suggested_answer_id')
-            if suggested_answer_id:
-                vals.update({'answer_score': self.env['survey.question.answer'].browse(int(suggested_answer_id)).answer_score})
-        return super(SurveyUserInputLine, self).create(vals_list)
-
-    def write(self, vals):
-        suggested_answer_id = vals.get('suggested_answer_id')
-        if suggested_answer_id:
-            vals.update({'answer_score': self.env['survey.question.answer'].browse(int(suggested_answer_id)).answer_score})
-        return super(SurveyUserInputLine, self).write(vals)
```

Comment:
I think we should put back the computation of the score in the create/write methods.
Especially now that there is a `_get_answer_score` method in the `survey.user_input.line` model (since a the recent task about survey sessions).

I know I mislead you on this, sorry about that.

Reply:
I put score computation back in create/write methods (directly into the the `_get_answer_score` method which I renamed to `_get_answer_score_values` since we still need to calculate non-choice question's `answer_is_correct`). Not sure if it's bad practice to pass in the full `vals` parameter. Let me know if I should just pass in all the used `vals` values directly rather than the whole thing (this means 4 arguments + `compute_speed_score`). I also adjusted the answer speed score calculation logic / corresponding docstring a bit for what I think simplifies it a bit. Let me know if I should split out this calculation logic into its own method as well.